iroh-gossip/src/net.rs

```diff
@@ -229,10 +231,14 @@ impl Gossip {
     ///
     /// This will be sent to peers on Neighbor and Join requests so that they can connect directly
     /// to us.
+    ///
+    /// This is only best effort, and will drop new events if backed up.
```

Comment:
That's why I opted for `Watch` in the first place: We only ever care for the latest value that was sent. Isn't there another alternative to `Watch` that keeps that behavior? Could just be something like a ringbuf with capacity 1.
But also fine to keep as is for now, there won't be more than 64 endpoint updates before the actor pulls them out.

Reply:
I don't trust watch.. I also looked at the tokio issue tracker and there are way too many cases where subtle misuse of it results in deadlocks